tests/integration/search_test.t

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use LWP::Simple "get";
+use Data::Dumper;
+use Encode;
+use File::Basename "dirname";
+use JSON::PP;
+use ProductOpener::TestDefaults qw/:all/;
+use ProductOpener::Test qw/:all/;
+use Getopt::Long;
+
+my $test_name = "search_test";
+my $tests_dir = dirname(__FILE__);
+my $expected_dir = $tests_dir . "/expected_test_results/" . $test_name;
+
+my %product_fields_1 = (
```

Comment:
Instead of m%products_fields_1, 2, 3 etc. you could declare an array of products.

e.g.

my @products = ( 
{
},
{
}
...

That way it's easy to add more products without having to declare new variables.

Reply:
Did you mean an array of all these hashmaps?